bin/MadGraph5_aMCatNLO/cards/production/13TeV/ZZTo4l0To3jets/ZZNJets_Zll_VBF_LO_proc_card.dat

```diff
@@ -0,0 +1,14 @@
+import model sm-ckm_no_b_mass
+
+define p = u c s d b u~ c~ s~ d~ b~ g
+define l+ = e+ mu+ ta+
+define l- = e- mu- ta-
+define j = p
+# Specify process(es) to run
+generate p p > z z @1
+add process p p > z z j @2
+add process p p > z z j j @3
+add process p p > z z j j j @4
```

Comment:
If you produce Z in ME level, it would give you to onshell Zs that would have very limited phase space. For your analysis, in RECO level, do you cut on mll just to be around the Z peak for both Zs? What you would have with this card will be two Zs that looks like slide12 top right in [1].

[1] https://indico.cern.ch/event/234296/contributions/1547084/attachments/391432/544354/13_02_16_tutomg.pdf

Reply:
@sihyunjeon  exactly. Nevertheless, the process with one of Z offshell (see bellow, to cover broader phase space) crashed due to computing issues while trying to decay the Z ( It is documented elsewhere in many HN msgs changed that time. These gridpacks were tough to be produced in slc6)

generate p p > z l+ l- @1
add process p p > z l+ l- j @2
add process p p > z l+ l- j j @3
add process p p > z l+ l- j j j @4

Reply:
I think it's because 2->6 decay is too heavy in your case. Anyhow with this syntax, you'll only get the phase space where mlls are very restricted "mll : 80-100". If your analysis is fine with this, it won't be a problem (== if your analysis look for two onshell Z bosons only)

Reply:
@sihyunjeon and @agrohsje , is it possible to break the generation in two gridpacks , leaving the heavy subprocess with 3 jets multiplicity to be generated separately ?

Reply:
@sznajder It is possible,
So you mean dividing it to two gridpacks?
1) p p > z l+ l- & p p > z l+ l- j  & p p > z l+ l- j j
2) p p > z l+ l- j j j

The merging would be handled purely in Pythia as long as the xqcut and mlm settings are correctly turned on so you don't need to change anything here.

But to synchronise with other physics process's gridpack settings, I think dividing it in to 4 gridpacks (0j, 1j, 2j, 3j) just like other jet binned processes would be better. What do you think?

Reply:
Yes, that's what I meant.  Then I think it's worth a try to generate the gridpacks separately and check if Madgraph can handle p p > z l+ l- j j j and Madspin with Z>l+l- separetely.
By the way , in this case one must have different PYTHIA matching setting for 'JetMatching:nJetMin' and 'JetMatching:nJetMax', , both set to the given jet multiplicity , correct ?

Reply:
@sznajder I totally agree that it's worth a try :)
No the Pythia fragments should also be kept the same. If you attempt to merge 0j, 1j, 2j, 3j samples, all these 4 Pythia fragments should have nJetMax = 3.

Reply:
Yes , that's correct about nJetMax, but what about nJet ?

PYTHIA  manual says:
JetMatching:nJet   (default = -1; minimum = -1)
When JetMatching:exclusive = 2, nJet indicates the minimum number of additional light jets in the incoming process. This value may be set automatically.

Shouldn't it be set to the given gridpack multiplicity, which is the minimum number of jets in each gridpack ? Or Pythia gets this automatically from the LHE information ?

Reply:
As far as I am aware, yes they don't need to be specified (and other jet binned samples don't do it as well).

@agrohsje Please correct me if I am wrong

Reply:
Right. You just need to specify the maximum number of additional partons in the P8 fragment. Independent of technically splitting gridpacks or not. This njetmax and inclusive/exclusive is e.g. needed for Alpgen production.

Reply:
@sihyunjeon @agrohsje Now I managed to do the process "p p > z l+ l- j j j" decaying with madspin which is the heaviest subprocess of our main process(*). Now I just need to understand how should we merge these different processes which with different XS to obtain our main process? Is the merge should be handled in Pythia, and how? maybe you can give an example config file?

(*)generate p p > z l+ l- @1
add process p p > z l+ l- j @2
add process p p > z l+ l- j j @3
add process p p > z l+ l- j j j @4

Reply:
https://github.com/cms-sw/genproductions/blob/master/genfragments/ThirteenTeV/Hadronizer/Hadronizer_TuneCP5_13TeV_MLM_5f_max1j_LHE_pythia8_cff.py

You should change the nJetMax to 3. But you should do some careful studies on xqcut/qcut, checking the DJR plots.

Reply:
@sihyunjeon , sorry I don't ask now about the qcut, let me rewrite the question,
I mean now I generated the processes (*) as we agreed before, each one in a separate gridpack.
what I need to understand now,  at which step should we merge these processes in one process?  Is this should be handled in Pythia and how?
(*)
generate p p > z l+ l- @1
generate p p > z l+ l- j@1
generate p p > z l+ l- jj@1
generate p p > z l+ l- jjj@1

Reply:
You mean you want to merge below?
X > Y
X > Y j
X > Y j j
X > Y j j j

You have to have separate datasets with same Pythia fragments.
ZZto4Lep_0jet
ZZto4Lep_1jet
ZZto4Lep_2jet
ZZto4Lep_3jet

You cannot merge separately generated gridpacks and combine them into one dataset (actually there is, but it is highly depreciated).